runtime/src/main/java/com/flipkart/flux/config/FileLocator.java

```diff
@@ -42,9 +43,11 @@
 	 */
 	private static final Logger LOGGER = LogManager.getLogger(FileLocator.class);
 	
-	/** The root folder where Flux runtime is deployed*/
-	@Inject
-	private static String deploymentRoot;
+	/** The config root folder of Flux */
+	private static final String DEPLOYED_CONFIG_ROOT;
```

Comment:
sure we can - https://github.com/google/guice/wiki/Injections

However, I agree we shouldn't. We can replace this with a decent builder or to constructor based injection. I don't see it as a pressing issue, but if you feel strongly about it then you can go ahead and do that - it'll be a trivial change